src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -25,22 +26,60 @@
 
     $scope.loadInProgress = false;
     $scope.saveInProgress = false;
+    $scope.isLegacyTemplate = true;
 
-    $scope.saveHtmlRaw = saveHtmlRaw;
+    $scope.save = save;
     $scope.saveAndEditWithMseditor = saveAndEditWithMseditor;
 
     if ($routeParams["templateId"]) {
       load($routeParams["templateId"]);
     }
+    else {
+      $scope.isLegacyTemplate = !$rootScope.forceMsEditor;
+    }
 
     /*************************/
     /* Function declarations */
     /*************************/
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
+      return templates.getTemplate(templateId).then(function (template) {
+        switch (template.bodyType) {
+          case "empty":
+            $scope.isLegacyTemplate = false;
+            break;
+          case "mseditor":
+            $scope.isLegacyTemplate = false;
+            break;
+          case "rawHtml":
+            $scope.isLegacyTemplate = true;
+            break;
+        }
+        if($rootScope.forceMsEditor && !$scope.isLegacyTemplate)
+        {
+          return {
+            from_name: template.from_name,
+            from_email: template.from_email,
+            subject: template.subject,
+            id: template.id,
+            name: template.name,
+            body: null
+          };
+        }
+        else {
+        return templates.getTemplateBody(templateId).then(function (body) {
+          return {
+            from_name: template.from_name,
+            from_email: template.from_email,
+            subject: template.subject,
+            id: template.id,
+            name: body.name,
```

Comment:
I think that it is better using _template.name_ in place of _body.name_

Also, there should be a way to reuse the code. Maybe using a kind of _noop_ promise? Let me know if you need help on this.

Reply:
👍 Ok, let me check it, thx!